core/shared/src/main/scala/zio/IO.scala

```diff
@@ -675,6 +675,12 @@ object IO {
    */
   val none: UIO[Option[Nothing]] = ZIO.none
 
+  /**
+   *  @see See [[zio.ZIO.not]]
+   */
+  def not[R, E](effect: ZIO[R, E, Boolean]): ZIO[R, E, Boolean] =
```

Comment:
No `R` type parameter here and just accept and return an `IO[E, A]`. Similar thing for each of the other companion object methods so we return the more specific type.